### PR TITLE
Storyshots addon refactoring

### LINKED ISSUE #3754: Storyshots with Storybook 4 for React fails with Angular import.
### Framework Loaders tries to load Angular stuff in React environment

After updating from 4.0-alpha.4 to -alpha.9 I see the following happens:

```
Cannot find module '@storybook/angular/options' from 'loader.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:210:17)
      at Object.<anonymous> (node_modules/@storybook/addon-storyshots/dist/angular/loader.js:7:16)
```

This is the not touched `Story.test.js` which is used to execute the test runner:

```js
import initStoryshots, {
  multiSnapshotWithOptions
} from "@storybook/addon-storyshots"
import { dirname } from "path"

initStoryshots({
  // Delegate to centralized config
  configPath: dirname(require.resolve("edge-storybook")),

  // Storing seperate snapshots for each individual story
  test: multiSnapshotWithOptions({}),

  // Ignore all containers, only snapshot pure components
  // without any app logic or data fetching
  storyKindRegex: /^((?!container).)*$/i
})

```

### Steps to reproduce

Executing `jest` on the command line.

### Please specify which version of Storybook and optionally any affected addons that you're running

- storybook/react 4.0.0-alpha.9
- storybook/addon-actions 4.0.0-alpha.9
- storybook/addon-links 4.0.0-alpha.9
- storybook/addon-storyshots 4.0.0-alpha.9
- storybook/addon-viewport 4.0.0-alpha.9
- jest 23.1.0

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: including changed needed for #2517 (probably for #3152 too)

This PR is rebased on "storyshots-separation" branch because I didn't want to wait for the #3742 to be merged. 

## What I did
1. Created and exposed a class called `Stories2SnapsConverter` in order to map story-files to snapshot-files and vise versa.
2. Refactored a structure of the addon to make it more maintainable/readable.
3. Fixed #3754

### CODE REVIEW

**Comment by Hypnosphi** [comment]
LMK when it's ready for review

**Comment by igor-dv** [comment]
I think it's ready. Will work on other stuff in smaller PRs.

File: addons/storyshots/storyshots-core/README.md [hunk 1]
```diff
@@ -307,9 +307,32 @@ initStoryshots({
 
 Take a snapshot of a shallow-rendered version of the component. Note that this option will be overriden if you pass a `renderer` option.
 
-### `getSnapshotFileName`
```
Comment (Hypnosphi) [inline addons/storyshots/storyshots-core/README.md]: It's probably worth mentioning in `MIGRATION.md`, as well as puppeteer moving to other package

**Comment by igor-dv** [comment]
@Hypnosphi, can I merge this?

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**